### PR TITLE
Support Swift 6

### PR DESCRIPTION
**Pull request by 0xTim**
Fully support Swift 6:

* Adds a Swift 6 manifest
* Fixes the last Sendable warning
* Fixes a deprecation
* Updates supported Swift version to 5.8

### CODE REVIEW

File: Sources/Vapor/HTTP/Headers/HTTPHeaderCacheControl.swift [hunk 1]
```diff
@@ -177,6 +177,27 @@ extension HTTPHeaders {
...
+            "no-cache": \.noCache,
+            "no-store": \.noStore,
+            "no-transform": \.noTransform,
+            "public": \.isPublic,
+            "private": \.isPrivate,
+            "proxy-revalidate": \.proxyRevalidate,
+            "only-if-cached": \.onlyIfCached
+        ]
+
+        private static let prefix: [String: WritableKeyPath<Self, Int?> & Sendable] = [
```
Comment (gwynne) [inline Sources/Vapor/HTTP/Headers/HTTPHeaderCacheControl.swift:193]: Might want to get rid of my junkier workaround at the bottom of the file if this works.

File: Sources/Vapor/Content/PlaintextEncoder.swift [hunk 2]
```diff
@@ -73,7 +73,9 @@ private final class _PlaintextEncoder: Encoder, SingleValueEncodingContainer {
     func encode<T>(_ value: T) throws where T: Encodable {
         if let data = value as? Data {
             // special case for data
-#if swift(>=5.7.2)
+#if swift(>=6)
+            let utf8Maybe = data.withUnsafeBytes({ $0.withMemoryRebound(to: CChar.self, { String(validatingCString: $0.baseAddress!) }) })
+#elseif swift(>=5.7.2)
```
Comment (gwynne) [inline Sources/Vapor/Content/PlaintextEncoder.swift:78]: Please just bump the base manifest to 5.8 already.

**Review by gwynne — CHANGES_REQUESTED** [review]
per comments

**Comment by 0xTim** [comment]
We're not going to introduce a 6.0 manifest for Vapor 4

**Pull request closed without merging**